DataDog/datadog-agent
# make sure domain strings are only sent once

### What does this PR do?

consolidates all domain strings into a single list, so that duplicates are not sent

Supercedes #8530 
### Motivation

more efficient packaging of submission data

### Testing instructions
Install at least one client, with dns tracking enabled.
Ensure data appears in-app as expected.

There's not a specific test, per-se as this shouldn't change the end result.
Testing is to ensure that there's no loss of data/regression.

### Checklist
